### PR TITLE
add manual page crop for pdf/djvu documents

### PR DESCRIPTION
**Pull request by chrox**

### CODE REVIEW

File: frontend/ui/reader/readercropping.lua [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	UIManager.repaint_all = true
+end
+
+function BBoxWidget:onCancelCrop()
+	UIManager:close(self)
+	self.ui:handleEvent(Event:new("SetZoomMode", self.orig_zoom_mode))
+	self.document.configurable.text_wrap = self.orig_reflow_mode
+	UIManager.repaint_all = true
+end
+
```
Comment (houqp) [inline frontend/ui/reader/readercropping.lua:2]: shall we put all the BBoxWdiget related definitions into a bboxwidget.lua file? That will make the structure more clear and if other widget want to use BBoxWdiget, they don't need to include readercropping.lua.

Comment (chrox) [inline frontend/ui/reader/readercropping.lua:2]: Sounds great. Patch appended.

File: frontend/ui/bbox.lua [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+					range = Geom:new{
+						x = 0, y = 0,
+						w = Screen:getWidth(),
+						h = Screen:getHeight()
+					}
+				}
+			},
+			CancelCrop = {
+				GestureRange:new{
+					ges = "hold_release",
```
Comment (houqp) [inline frontend/ui/bbox.lua:88]: will be better if we use "hold" here, so use can see the effect once hold event is fired.

Comment (chrox) [inline frontend/ui/bbox.lua:88]: Yes. "hold" should be more natural and reasonable. I will send another PR on this issue.

**Pull request merged**